src/Public/js/nipsweb.import.js

```diff
@@ -133,45 +134,56 @@ function updateTimeslotList() {
 
 // Stage 4) Select the timeslot & load the showplan from it.
 $("#import-timeslot-selector").change(function() {
-  loadChannelList();
+  loadShowPlan();
   $("#import-showplan").fadeIn();
 });
 
-function loadChannelList() {
+function loadShowPlan() {
   var selectedTimeslotID = $("#import-timeslot-selector").find(":selected").attr("value");
-  myradio.callAPI("GET","timeslot","showplan",selectedTimeslotID,"","",
+  myradio.callAPI("GET", "timeslot", "showplan", selectedTimeslotID, "", "",
     function (data) {
       for (item in data) {
         if (item === "myradio_errors") {
           continue;
         }
       }
-      $("#import-channel-list").empty();
-      var selectedChannelNo = $("#import-channel-selector a.active").attr("channel");
-      var item, itemid, cleanStars, extraString;
-      for (item in data.payload[selectedChannelNo]) {
-        if (data.payload[selectedChannelNo][item].type == "central") {
-          if (!data.payload[selectedChannelNo][item].clean) {
-            cleanStars = "**";
-          } else {
-            cleanStars = "";
-          }
-          extraString = " - " + data.payload[selectedChannelNo][item].artist + " - " + data.payload[selectedChannelNo][item].album.title + " (" + data.payload[selectedChannelNo][item].length + ")";
-          itemid = data.payload[selectedChannelNo][item]["album"].recordid + "-" + data.payload[selectedChannelNo][item].trackid;
+      currentShowPlan = data.payload;
```

Comment:
Global state :(
How feasible would it be to make this a parameter for loadChannelList ?

Reply:
Not sure what to here, apart from reloading the show plan AJAX every time I change channel (which I fixed in this PR from previous behavior)

Reply:
Without creating all the channels and then dynamically hiding and showing them, I think this is as good as we are going to get

Reply:
Yeah.